File: src/plugins/hello.lua
return {
  name = "hello-lua",
  version = "0.1.0",
  setup = function(a)
    a.addCommand("hellolua", function()
      a.log("Hello from Lua plugin!")
    end)

    a.on("editor:ready", function()
      a.log("hello-lua plugin ready")
    end)
  end
}

File: src/plugins/hello.ts
import type { PluginDefinition } from "../plugin/types";

const plugin: PluginDefinition = {
  name: "hello-ts",
  version: "0.1.0",
  setup(api) {
    api.addCommand("hello", () => {
      api.log("Hello from TypeScript plugin!");
    });

    api.on("editor:ready", () => {
      api.log("hello-ts plugin ready");
    });
  },
};

export default plugin;